pyo3-macros-backend/src/pyclass.rs

```diff
@@ -387,6 +393,19 @@ fn impl_class(
     let Ctx { pyo3_path, .. } = ctx;
     let pytypeinfo_impl = impl_pytypeinfo(cls, args, ctx);
 
+    if let Some(str) = &args.options.str {
+        if str.value.is_some() {
+            // check if any renaming is present
+            let renaming_conflict = field_options.iter().all(|x| x.1.name.is_none())
+                & args.options.name.is_none()
+                & args.options.rename_all.is_none();
+            ensure_spanned!(renaming_conflict, str.value.span() => "The optional string format shorthand argument to `str` is incompatible with any renaming via `name` or `rename_all`.  You should remove the string format shorthand argument and implement the `Display` trait or implement `__str__` directly.");
```

Comment:
This reads a bit funny, since `renaming_conflict` is true when there is no conflict 😄 
Maybe you turn the logic around here and then ensure `!renaming_conflict`? Or you just rename the binding.

Reply:
lol!  Yes,  I originally had the inverse logic here and reversed it for clarity.  So much for clarity since I forgot to rename this!  I will rename the binding.